### PR TITLE
Add regression transformer

### PR DESCRIPTION
**Pull request by Pankhil07**

### CODE REVIEW

**Review by AdamIzdebski — CHANGES_REQUESTED** [review]

File: configs/models/regression_transformer/config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "model_name": "RegressionTransformer",
+    "temperature": 1.0, 
+    "fraction_to_mask": 0.4
+}
+
```
[the remark this answers is not in the document]
Comment (AdamIzdebski) [reply configs/models/regression_transformer/config.json]: done

File: jointformer/models/auto.py [hunk 2]
```diff
@@ -32,6 +32,11 @@ def from_config(cls, config: ModelConfig) -> BaseModel:
             return getattr(importlib.import_module(
                 "jointformer.models.moler"),
                 "Moler").from_config(config)
-            
+        
+        if config.model_name == "RegressionTransformer":
+            return getattr(importlib.import_module(
+                "jointformer.models.moler"),
+                "Moler").from_config(config)
```
[the remark this answers is not in the document]
Comment (AdamIzdebski) [reply jointformer/models/auto.py]: done

File: jointformer/models/regression_transformer.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        self._temperature = None
+        self._top_k = None
+        self._device = None
+        self._tolerence  = None
+        self._algorithm_version = None
+        self._search_strategy = None 
+        self._target = None
+        self._fraction_to_mask = None
+
+    def to_guacamole_generator(self, temperature, fraction_to_mask, device, target, *args, **kwargs) -> DistributionMatchingGenerator:
```
[the remark this answers is not in the document]
Comment (AdamIzdebski) [reply jointformer/models/regression_transformer.py]: done

File: jointformer/models/regression_transformer.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        pass #TODO: implement
+        # rets = []
+        # for i in tqdm(range(0, len(smiles), self._batch_size), "Encoding samples"):
+        #     batch = smiles[i:i+self._batch_size]
+        #     enc = _encode_from_smiles(self._dataset, self._model, batch)
+        #     rets.extend(enc)
+        # return np.stack(rets, axis=0)
+
+    def load_pretrained(self, filename, *args, **kwargs):
+        self._dataset = self._load_regression_transformer_dataset(filename)
```
[the remark this answers is not in the document]
Comment (AdamIzdebski) [reply jointformer/models/regression_transformer.py]: done

File: jointformer/configs/model.py [hunk 5]
```diff
@@ -19,6 +19,8 @@ def __init__(
         head_dim: Optional[int] = None,
         num_layers: Optional[int] = None,
         bias: Optional[bool] = None,
+        temperature: Optional[int] = None,
+        fraction_to_mask:Optional[float] = None,
```
Comment (AdamIzdebski) [inline jointformer/configs/model.py:23]: to delete

File: jointformer/configs/model.py [hunk 6]
```diff
@@ -44,6 +46,8 @@ def __init__(
         self.head_dim = head_dim
         self.num_layers = num_layers
         self.bias = bias
+        self.temperature = temperature
+        self.fraction_to_mask = fraction_to_mask
```
Comment (AdamIzdebski) [inline jointformer/configs/model.py:50]: to delete

**Review by AdamIzdebski — APPROVED** [review]

**Review by jano1906 — APPROVED** [review]

**Pull request merged**